packages/cadl-ranch-specs/http/resiliency/srv-driven/old.tsp

```diff
@@ -0,0 +1,148 @@
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the first service spec can call the first deployment of a service with api version v1
+  - A client generated from the first service spec can call the second deployment of a service with api version v1
+
+  In the next service spec, we will test that we can grow this operation from accepting one optional parameter to accepting two optional parameters.
+""")
+  @route("/from-one-optional")
+  @doc("Test that currently accepts one optional parameter, will be updated in next spec to accept a new optional parameter as well")
+  @get
+  fromOneOptional(
+    @doc("I am an optional parameter")
+    @query
+    parameter?: string
+  ): NoContentResponse;
+}
+
+@scenario
+@scenarioDoc("""
+  Need the following two calls:
+  - Pass in `serviceDeploymentVersion="v1"` and `apiVersion="v1"` with header `content-type="application/json"` and expected request body:
+  ```json
+  { url: \"http://example.org/myimage.jpeg\" }
+  ```
+  - Pass in `serviceDeploymentVersion="v2"` and `apiVersion="v1"` with header `content-type="application/json"` and expected request body:
+  ```json
+  { url: \"http://example.org/myimage.jpeg\" }
+  ```
+
+  There are three concepts that should be clarified:
+  1. Client spec version: refers to the spec that the client is generated from. 'v1' is a client generated from old.tsp and 'v2' is a client generated from main.tsp.
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the first service spec can call the first deployment of a service with api version v1
+  - A client generated from the first service spec can call the second deployment of a service with api version v1
+
+  In the next service spec, we will test that we can grow this operation from accepting one content type to two.
+""")
+@route("/add-content-type")
+@doc("Test that currently accepts one content type. Will be updated in next spec to accept a new content type as well.")
+@post
+op addContentType(
+  @path contentTypePath: "json",
```

Comment:
This path param is weird. I think for the first version, we should remove this path param. For the evolutional version, we shall define two operations with shared route (I don't know what is the final decision of the best practice for overload):
```
@scenario
@scenarioDoc("""
Operation only accept application/json content. After evolution, this operation could accept both application/json and image/jpeg.

Expected header: contentType=application/json
Expected request body:
```json
{
  "message": "hello"
}
Expected response body:
```json
{
  "message": "hello"
}
""")
@route("/contentType")
@post
op contentType(
  @body
  parameter: Message
): Message;

@scenario
@scenarioDoc("""
Operation only accept application/json content. After evolution, this operation could accept both application/json and image/jpeg.

Expected header: contentType=image/jpeg
Expected request body: binary date of the image
Expected response body:
```json
{
  "message": "hello"
}
""")
@route(
  "/contentType",
  {
    shared: true,
  }
)
@post
op contentTypeOfImage(
  @body
  parameter: bytes
): Message;
```